### PR TITLE
Update CI images to be centos and build a dedicated small images for unit-tests and lint

### PR DESCRIPTION
**Pull request by michaellevy101**

### CODE REVIEW

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml [hunk 1]
```diff
@@ -23,6 +23,10 @@ base_images:
     name: stream8
     namespace: coreos
     tag: "8"
+  coreos_centos_stream9:
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:26]: be aware it does not use it currently as there was no image substitution in the building of ``project_image``

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml [hunk 2]
```diff
@@ -48,11 +52,17 @@ base_images:
...
-    name: release
-    namespace: openshift
-    tag: golang-1.17
+  project_image:
+    dockerfile_literal: |
+      FROM quay.io/centos/centos:stream9
+      RUN dnf install -y 'dnf-command(config-manager)' && \
+          dnf config-manager --set-enabled crb && \
+          dnf install -y git postgresql-server unzip make
+      COPY --from=registry.ci.openshift.org/openshift/release:golang-1.17 /usr/local/go /usr/local/go
```
[the remark this answers is not in the document]
Comment (osherdp) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:57]: That's not the scope that I'm referring to. This current PR affecting much than unit-test job. This is the aspect that I think is too much out-of-scope

[the remark this answers is not in the document]
Comment (michaellevy101) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml]: that's ok. It should affect all our images to be centos9. I updated the PR description.

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml [hunk 3]
```diff
@@ -53,6 +57,44 @@ build_root:
...
+- dockerfile_literal: |
+    FROM base
+    COPY --from=quay.io/app-sre/golangci-lint:v1.37.1 /usr/bin/golangci-lint /usr/bin/golangci-lint
+  from: assisted-service-centos9
+  to: assisted-service-lint
+- dockerfile_literal: |
+    FROM base
+    RUN make build-openshift-ci-test-bin
+  from: assisted-service-centos9
+  to: assisted-service-test
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:97]: let's rename to ``assisted-service-subsystem`` to make it 100% clear

**Review by osherdp — CHANGES_REQUESTED** [review]
just one little thing

**Review by osherdp — APPROVED** [review]

**Pull request merged**